services/movie_session.py

```diff
@@ -0,0 +1,52 @@
+from datetime import datetime
+from typing import Optional
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> None:
+    MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id
+    )
+
+
+def get_movies_sessions(session_date: str = None) -> QuerySet:
+    sessions = MovieSession.objects.all()
+    if session_date:
+        if session_date:
+            sessions = sessions.filter(show_time__date=session_date)
+    return sessions
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(pk=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: Optional[datetime] = None,
+        movie_id: Optional[int] = None,
+        cinema_hall_id: Optional[int] = None
+) -> None:
+    if not any([show_time, movie_id, cinema_hall_id]):
+        return
+    movie_session = get_movie_session_by_id(session_id)
```

Comment:
what if you create a dict, and then you will check if any of the optional arguments are passed to this function. In case if there are - you will add them to the dict. Then after you checked if your dict has some values to update you will call update on queryset, and if there are not any you don't do anything